### ISSUE TITLE
per property historyChangeEvent only

### ISSUE THREAD

**Issue by brod** [brod/mongoose-track#3]
Currently if **no properties** within an `Object` are `defined` and multiple properties within that document are changed within a single `document.save()` a single **historyChangeEvent** will occur for _all_ properties.

Ideally, the **historyChangeEvent** would be per property outside of the initial document creation.

```js
let fruitSchema = new mongoose.Schema({
    name: { type: String },
    color: {
        primary: { type: String },
        secondary: { type: String }
    }
})

fruitSchema.plugin(mongooseTrack.plugin)

let fruitModel = mongoose.model('fruitModel', fruitSchema)

let tempFruit = new fruitModel({
    name: 'Banana'
})

tempFruit.save()
    .then(function(fruitDocument){
        fruitDocument.color.primary = 'yellow'
        fruitDocument.color.secondary = 'red'
        return fruitDocument.save()
    })
    .then(function(fruitDocument){
        console.log(fruitDocument.history[0])

        /*
        Currently:
        [{
            path: ['color'],
            before: undefined,
            after: { primary: 'yellow', secondary: 'red' }
        }]

        Ideally:
        [{
            path: ['color', 'primary'],
            before: undefined,
            after: 'yellow'
        },{
            path: ['color', 'secondary'],
            before: undefined,
            after: 'red'
        }]
        */
    })
```

**Comment by brod** [comment]
This isn't really a bug, in the example above, `document.color` was `undefined` and the **historyChangeEvent** was created as intended, but it'd be nice to have more verbose **historyChangeEvent**'s.
